api/src/main/java/org/apache/iceberg/actions/compaction/DataCompactionStrategy.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions.compaction;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.Table;
+
+interface DataCompactionStrategy extends Serializable {
+  /**
+   * Returns the name of this compaction strategy
+   */
+  String name();
+
+  /**
+   * Returns a set of options which this compaction strategy can use. This is an allowed-list and any options not
+   * specified here will be rejected at runtime.
+   */
+  Set<String> validOptions();
+
+  DataCompactionStrategy withOptions(Map<String, String> options);
+
+  /**
+   * Selects files which this strategy believes are valid targets to be rewritten.
+   *
+   * @param dataFiles iterator of live datafiles in a given partition
+   * @return iterator containing only files to be rewritten
+   */
+  Iterable<FileScanTask> selectFilesToCompact(Iterable<FileScanTask> dataFiles);
+
+  /**
+   * Groups file scans into lists which will be processed in a single executable unit. Each group will end up being
+   * committed as an independent set of changes. This creates the jobs which will eventually be run as by the underlying
+   * Action.
+   *
+   * @param dataFiles iterator of files to be rewritten
+   * @return iterator of sets of files to be processed together
+   */
+  Iterable<Set<FileScanTask>> groupFilesIntoChunks(Iterable<FileScanTask> dataFiles);
```

Comment:
just `groupFiles`? or `planFileGroups`?

Reply:
Can we use `Set` here? it doesn't seem that `BaseFileScanTask` implements any `hashCode` or `equals` method.

Reply:
since Iceberg's `TableScan` returns `CloseableIterable`, should we use it here too (instead of the regular `Iterable`).  if we shouldn't use `Set`, maybe we just use `CombinedScanTask`. so basically returns `CloseableIterable<CombinedScanTask>`?

Reply:
This was originally List but was changed to set to accommodate a previous reviewer's request. I think you're correct and we should switch it back to list. I'll also do a rename to planFileGroups

Reply:
I don't think we should use CloseableIterable, I think it adds a bit of complexity to the code which we won't really benefit from.

Reply:
agree with you on Iterable. maybe just use `CombinedScanTask` interface and return `Iterable<CombinedScanTask>`. Each CombinedScanTask contains a group of files for rewrite action?

Reply:
It's better the keep it as FileScanTasks if you see the work we've been doing on 
FileScanTaskSetManager  https://github.com/apache/iceberg/pull/2472
and SparkRewriteManger https://github.com/apache/iceberg/pull/2500

Reply:
ok. I just thought Iceberg already had an interface `CombinedScanTask` to capture of a group of files. Reuse it might make make sense in the `api` module.